src/moj/components/date-picker/date-picker.js

```diff
@@ -0,0 +1,596 @@
+  const date = new Date(this.currentDate)
+  date.setDate(date.getDate() - 1)
+  this.goToDate(date)
+}
+
+// week navigation
+Datepicker.prototype.focusNextWeek = function () {
+  const date = new Date(this.currentDate)
+  date.setDate(date.getDate() + 7)
+  this.goToDate(date)
+}
+
+Datepicker.prototype.focusPreviousWeek = function () {
+  const date = new Date(this.currentDate)
+  date.setDate(date.getDate() - 7)
+  this.goToDate(date)
+}
+
+Datepicker.prototype.focusFirstDayOfWeek = function () {
+  const date = new Date(this.currentDate)
+  date.setDate(date.getDate() - date.getDay())
+  this.goToDate(date)
+}
+
+Datepicker.prototype.focusLastDayOfWeek = function () {
+  const date = new Date(this.currentDate)
+  date.setDate(date.getDate() - date.getDay() + 6)
+  this.goToDate(date)
+}
+
+// month navigation
+Datepicker.prototype.focusNextMonth = function (event, focus = true) {
+  event.preventDefault()
+  const date = new Date(this.currentDate)
+  date.setMonth(date.getMonth() + 1, 1)
+  this.goToDate(date, focus)
+}
+
+Datepicker.prototype.focusPreviousMonth = function (event, focus = true) {
+  event.preventDefault()
+  const date = new Date(this.currentDate)
+  date.setMonth(date.getMonth() - 1, 1)
+  this.goToDate(date, focus)
+}
+
+// year navigation
+Datepicker.prototype.focusNextYear = function (event, focus = true) {
+  event.preventDefault()
+  const date = new Date(this.currentDate)
+  date.setFullYear(date.getFullYear() + 1, date.getMonth(), 1)
+  this.goToDate(date, focus)
+}
+
+Datepicker.prototype.focusPreviousYear = function (event, focus = true) {
+  event.preventDefault()
+  const date = new Date(this.currentDate)
+  date.setFullYear(date.getFullYear() - 1, date.getMonth(), 1)
+  this.goToDate(date, focus)
+}
+
+/**
+ *
+ * @param button
+ * @param index
+ * @param row
+ * @param column
+ * @param picker
+ * @constructor
+ */
+function DSCalendarDay(button, index, row, column, picker) {
+  this.index = index
+  this.row = row
+  this.column = column
+  this.button = button
+  this.picker = picker
+
+  this.date = new Date()
+}
+
+DSCalendarDay.prototype.init = function () {
+  this.button.addEventListener('keydown', this.keyPress.bind(this))
+  this.button.addEventListener('click', this.click.bind(this))
+}
+
+DSCalendarDay.prototype.update = function (day, hidden, disabled) {
+  this.button.innerHTML = day.getDate()
+  this.date = new Date(day)
+
+  if (disabled) {
+    this.button.setAttribute('disabled', true)
+  } else {
+    this.button.removeAttribute('disabled')
+  }
+
+  if (hidden) {
+    this.button.style.display = 'none'
+  } else {
+    this.button.style.display = 'block'
+  }
+}
+
+DSCalendarDay.prototype.click = function (event) {
+  this.picker.goToDate(this.date)
+  this.picker.selectDate(this.date)
+
+  event.stopPropagation()
+  event.preventDefault()
+}
+
+DSCalendarDay.prototype.keyPress = function (event) {
+  let calendarNavKey = true
+
+  switch (event.keyCode) {
```

Comment:
`keyCode` is deprecated, but the replacement [code](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code#browser_compatibility) doesn't have sufficient support (most notably, Chrome for Android doesn't support it)

Reply:
Based on the link, support looks pretty good across the board.  Chrome for Android looks to have had support from 2016.  Maybe the page was incorrect and has since been updated.